pants

```diff
@@ -2,6 +2,39 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P)
+
+_interpreter_version_sentinel_file_path='.pants.d/.pants-python-version'
+
+function clean_if_interpreter_changed() {
+  if [[ -n "${_recursive_pants_shell_script_invocation:-}" ]]; then
+    return 0
+  fi
+
+  _python_version_file="${REPO_ROOT}/${_interpreter_version_sentinel_file_path}"
+
+  if [[ -f "$_python_version_file" ]]; then
+    _prev_run_python_version="$(cat "$_python_version_file")"
+    if [[ "$PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS" == "$_prev_run_python_version" ]]; then
+      return 0
+    fi
+  else
+    cat >&2 <<EOF
+${_interpreter_version_sentinel_file_path} not found in the buildroot.
```

Comment:
```suggestion
${interpreter_constraints_file} not found in the buildroot.
```

Reply:
I kept this as `${interpreter_constraints_sentinel_file_path}` as per the previous conversation we had about the full path name being unnecessary here.